proto/prost/private/protoc_wrapper.rs

```diff
@@ -0,0 +1,1047 @@
+//! A process wrapper for running a Protobuf compiler configured for Prost or Tonic output in a Bazel rule.
+
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use std::fmt::{Display, Formatter, Write};
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process;
+use std::{env, fmt};
+
+use prost::Message;
+use prost_types::{
+    DescriptorProto, EnumDescriptorProto, FileDescriptorProto, FileDescriptorSet,
+    OneofDescriptorProto,
+};
+
+/// Locate prost outputs in the protoc output directory.
+fn find_generated_rust_files(out_dir: &Path) -> BTreeSet<PathBuf> {
+    let mut all_rs_files: BTreeSet<PathBuf> = BTreeSet::new();
+    for entry in fs::read_dir(out_dir).expect("Failed to read directory") {
+        let entry = entry.expect("Failed to read entry");
+        let path = entry.path();
+        if path.is_dir() {
+            for f in find_generated_rust_files(&path) {
+                all_rs_files.insert(f);
+            }
+        } else if let Some(ext) = path.extension() {
+            if ext == "rs" {
+                all_rs_files.insert(path);
+            }
+        } else if let Some(name) = path.file_name() {
+            if name == "_" {
```

Comment:
When does this happen? Can you add a comment?

Reply:
I believe this comes from https://github.com/neoeinstein/protoc-gen-prost/blob/protoc-gen-prost-v0.2.2/protoc-gen-prost/src/lib.rs#L69

Reply:
That's correct, and I double checked that this is absolutely necessary. It shows up specifically when the package name is empty. I have added a comment.